src/main/java/au/org/democracydevelopers/raireservice/persistence/entity/Assertion.java

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2024 Democracy Developers
+
+The Raire Service is designed to connect colorado-rla and its associated database to
+the raire assertion generation engine (https://github.com/DemocracyDevelopers/raire-java).
+
+This file is part of raire-service.
+
+raire-service is free software: you can redistribute it and/or modify it under the terms
+of the GNU Affero General Public License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+raire-service is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.raireservice.persistence.entity;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import jakarta.persistence.*;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * RAIRE generates a set of assertions for a given IRV contest. The different types of assertion
+ * that RAIRE can generate are defined as subclasses of this base Assertion class. For a description
+ * of what assertions are and the role they play in an IRV audit, see the Guide to RAIRE.
+ */
+@Entity
+@Table(name = "assertion")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "assertion_type")
+public abstract class Assertion {
+
+  protected static final Logger logger = LoggerFactory.getLogger(Assertion.class);
+
+  /**
+   * Assertion ID.
+   */
+  @Id
+  @Column(updatable = false, nullable = false)
+  @GeneratedValue(strategy = GenerationType.SEQUENCE)
```

Comment:
Based on your suggestion, I agree that GenerationType.IDENTITY will be better for this case.

Reply:
Worth adding a test that it works with the issue you mentioned, i.e. that pre-filled database records didn't count for the increment.